include/Board.h

```diff
@@ -0,0 +1,33 @@
+#pragma once
+
+#include <memory>
+#include <map>
+#include <vector>
+#include <iostream>
+
+#include "PieceFactory.h"
+
+class Piece;
+
+class Board {
+public:
+    static Board& getInstance();
+    int move(const std::string&);
+    std::pair<std::pair<int, int>, std::pair<int, int>> parseMoveString(const std::string&) const;
+    bool isValidPosition(int, int) const;
+    void fillBoard(const std::string&);
+
+private:
+    bool m_currentPlayer = 1;
+    static constexpr int m_numRows = 8;
+    static constexpr int m_numCols = 8;
+
+    std::vector<std::vector<std::unique_ptr<Piece>>> m_chessMatrix;
```

Comment:
Instead of using raw pointers, consider using smart pointers (e.g., std::unique_ptr or std::shared_ptr) to manage ownership and ensure proper resource management.

Reply:
Yet I have used unique_ptrs. Please clarify if I'm missing something.

Reply:
No, good job